python/ray/autoscaler/_private/commands.py

```diff
@@ -678,6 +678,9 @@ def get_or_create_head_node(config: Dict[str, Any],
         if not no_restart:
             warn_about_bad_start_command(ray_start_commands)
 
+        if config["provider"]["type"] == "aws":
```

Comment:
Since we're adding update_node to the node_provider interface, we can omit this if statement and always invoke the function. If non-aws providers don't have their own implementation, then it will just do nothing or, if they do have their own implementation in the future, then it can actually do useful work for their respective node provider implementations.